Modules/_hashopenssl.c

```diff
@@ -1754,7 +1785,8 @@ _hashlib_HMAC_digest_impl(HMACobject *self)
     unsigned char digest[EVP_MAX_MD_SIZE];
     unsigned int digest_size = _hmac_digest_size(self);
     if (digest_size == 0) {
-        return _setException(PyExc_ValueError, NULL);
+        assert(PyErr_Occurred());
```

Comment:
These asserts are not really needed, since the interpreter has a similar assert. If you return `NULL` from a C-implemented Python API without an exception set, the interpreter will crash in debug builds.